torch/csrc/distributed/autograd/init.cpp

```diff
@@ -75,7 +75,7 @@ PyObject* dist_autograd_init(PyObject* /* unused */) {
   module.def(
       "_release_context",
       [](int64_t context_id) {
-        return DistAutogradContainer::getInstance().releaseContext(context_id);
+        return DistAutogradContainer::getInstance().releaseContextIfPresent(context_id);
```

Comment:
Why are we changing this? This should be `releaseContext` and not `releaseContextIfPresent`.

Reply:
When `_release_context` is called from (https://github.com/pytorch/pytorch/blob/3bccd3fc0db8265bd26139069fdf1fcdac911b90/torch/distributed/autograd/*init*.py#L29) (the only place it's called from), it attempts to release contexts that have already been cleaned up by this watchdog thread, and thus throws an Exception. If we simply check if the contexts have already been released, which `releaseContextIfPresent` does, then we avoid this Exception. Is this ok? I can alternately create another function called `_release_context_if_present` in `torch/csrc/distributed/autograd/init.cpp`, which can be called from the `__exit__()` function linked above.

Reply:
This would not happen if we only clean up context ids originating from other workers.